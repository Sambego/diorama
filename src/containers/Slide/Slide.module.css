.slide {
  width: 100%;
  height: 100%;
}

.content {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  text-align: center;
}
