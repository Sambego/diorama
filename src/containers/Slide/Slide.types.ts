import { MultiSlideInfo } from "@contexts/MultiSlideContext";
import React from "react";

export type SlideProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
	notes?: string;
	// eslint-disable-next-line react/require-default-props
	index?: number;
	// eslint-disable-next-line react/require-default-props
	navigate?: (slideIndex: number) => void;
}>;
export type SlideInfo = {
	notes?: SlideProps["notes"];
	multiSlideInfo?: MultiSlideInfo;
};
