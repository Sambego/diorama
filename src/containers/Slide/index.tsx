/* eslint-disable react/no-unused-prop-types */
import React, { useContext, useEffect } from "react";
import PropTypes from "prop-types";

import { DeckContextInternal } from "@contexts/DeckContext";
import styles from "./Slide.module.css";
import { SlideProps } from "./Slide.types";

function Slide({ children, style, className, notes }: SlideProps) {
	const intDeckCtx = useContext(DeckContextInternal);
	useEffect(() => {
		intDeckCtx.setSlideNotes(notes);
		return () => {
			intDeckCtx.setSlideNotes(null);
		};
	}, [notes]);
	return (
		<div style={style} className={`${styles.slide} diorama-slide ${className}`}>
			<div className={`diorama-content ${styles.content}`}>{children}</div>
		</div>
	);
}

Slide.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	notes: PropTypes.string,
	style: PropTypes.shape({}),
};

Slide.defaultProps = {
	className: "",
	notes: undefined,
	style: {},
};

export default Slide;
