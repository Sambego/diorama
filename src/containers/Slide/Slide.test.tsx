import React from "react";
import renderer from "react-test-renderer";
import Slide from ".";

it("renders correctly", () => {
	const tree = renderer.create(<Slide>Slide content</Slide>).toJSON();
	expect(tree).toMatchSnapshot();
});
