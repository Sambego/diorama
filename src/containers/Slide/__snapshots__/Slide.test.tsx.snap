// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="_slide_1aodf_1 diorama-slide "
  style={{}}
>
  <div
    className="diorama-content _content_1aodf_6"
  >
    Slide content
  </div>
</div>
`;
